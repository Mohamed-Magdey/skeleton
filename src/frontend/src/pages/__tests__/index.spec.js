import { mount } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import Page from '../index'

const vuetify = new Vuetify()

describe('index.vue', () => {
  let store

  beforeEach(() => {
    const actions = {
      'page/title': jest.fn(),
      'loading/start': jest.fn(),
      'loading/finish': jest.fn()
    }
    store = new Vuex.Store({
      actions,
      state: {
        loading: { status: false }
      }
    })
  })

  it('should match snapshot', () => {
    const router = new VueRouter({})
    const w = mount(Page, {
      store,
      router,
      stubs: ['router-link', 'router-view'],
      vuetify,
      mocks: {
        $t: msg => msg
      }
    })

    expect(w.html()).toMatchSnapshot()
    w.destroy()
  })
})
