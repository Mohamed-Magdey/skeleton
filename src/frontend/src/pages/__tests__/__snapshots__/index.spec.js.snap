// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue should match snapshot 1`] = `
<div>
  <section id="home"></section>
  <div class="main-wrap">
    <!---->
    <mq-layout mq="lgDown,xl">
      <div class="page-nav">
        <nav class="section-nav">
          <scrollactive offset="20" active-class="active" tag="section"><a href="#feature" class="anchor-link scrollactive-item" style="top: 30px;"><span class="v-tooltip v-tooltip--left"><!----><span>feature</span></span></a><a href="#pricing" class="anchor-link scrollactive-item" style="top: 0px;"><span class="v-tooltip v-tooltip--left"><!----><span>pricing</span></span></a></scrollactive>
        </nav> <span class="v-tooltip v-tooltip--left"><!----><scrollactive tag="div"><a href="#home" class="fab anchor-link scrollactive-item v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round theme--light v-size--default primary"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-up theme--dark"></i></span></a></scrollactive></span>
      </div>
    </mq-layout>
    <mq-layout mq="lgDown,xl">
      <div class="v-snack notification v-snack--bottom v-snack--has-background" style="padding-bottom: 0px; padding-top: 0px;">
        <transition-stub name="v-snack-transition">
          <div class="v-snack__wrapper v-sheet theme--dark" style="display: none;">
            <div role="status" aria-live="polite" class="v-snack__content">
              <div class="action">
                By continuing to use the site, you agree to the use of cookies. more information
              </div> <button type="button" class="button v-btn v-btn--outlined theme--dark v-size--default white--text"><span class="v-btn__content">
    common.accept
  </span></button>
            </div>
            <div class="v-snack__action "></div>
          </div>
        </transition-stub>
      </div>
    </mq-layout>
  </div>
</div>
`;
